test/funder.js

```diff
@@ -0,0 +1,328 @@
+ * @param {string} address Beneficiary address where funds will be added;
+ * @return {Promise<string>} Transaction hash returned by faucet.
+ */
+const fundAccountFromRopstenFaucet = address => new Promise(
+  (resolve, reject) => axios.post(ROPSTEN_FAUCET_URL, {
+    toWhom: address,
+  })
+    .then(response => resolve(response.data))
+    .catch(error => reject(error)),
+);
+
+/**
+ * This methods takes various faucet fund request as a promise and waits
+ * till all the promises are not resolved.
+ * @param {Array<Promise>} originMosaicFaucetFundRequests
+ * @param {Array<Promise>} auxiliaryMosaicFaucetFundRequests
+ * @param {Array<Promise>} ropstenFaucetFundRequest
+ * @param {Web3} originWeb3
+ * @param {Web3} auxiliaryWeb3
+ * @return {Promise<FundingDetails>}
+ */
+const waitForFunding = async (
+  originMosaicFaucetFundRequests,
+  auxiliaryMosaicFaucetFundRequests,
+  ropstenFaucetFundRequest,
+  originWeb3,
+  auxiliaryWeb3,
+) => Promise.all([
+  originMosaicFaucetFundRequests,
+  auxiliaryMosaicFaucetFundRequests,
+  ropstenFaucetFundRequest,
+]).then((faucetResponse) => {
+  const originTransactionHashes = faucetResponse[0].map(response => response.txHash);
+  const auxiliaryTransactionHashes = faucetResponse[1].map(response => response.txHash);
+  const ropstenTransactionHashes = faucetResponse[2].map(response => response.txHash);
+
+  return Promise.all([
+    originWeb3.eth.getTransactionReceiptMined(originTransactionHashes),
+    auxiliaryWeb3.eth.getTransactionReceiptMined(auxiliaryTransactionHashes),
+    originWeb3.eth.getTransactionReceiptMined(ropstenTransactionHashes),
+  ]).then(receipts => ({
+    receipts: {
+      originFaucetReceipts: receipts[0],
+      auxiliaryFaucetReceipts: receipts[1],
+      ropstenFaucetReceipts: receipts[2],
+    },
+    txHashes: {
+      originFaucetTXHashes: originTransactionHashes,
+      auxiliaryFaucetTXHashes: auxiliaryTransactionHashes,
+      ropstenFaucetTXHashes: ropstenTransactionHashes,
+    },
+  }));
+});
+
+/**
+ * This method returns receipt for a given transaction hash. It waits till
+ * transaction is mined.
+ * @param {Array | string} txHash Single or many transaction hashes.
+ * @param {Number=} interval Polling interval in milli second.
+ */
+function getTransactionReceiptMined(txHash, interval) {
+  const self = this;
+  const transactionReceiptAsync = (resolve, reject) => {
+    self.getTransactionReceipt(txHash, (error, receipt) => {
+      if (error) {
+        reject(error);
+      } else if (receipt == null) {
+        setTimeout(
+          () => transactionReceiptAsync(resolve, reject),
+          interval || 500,
+        );
+      } else {
+        resolve(receipt);
+      }
+    });
+  };
+
+  if (Array.isArray(txHash) && txHash.length > 0) {
+    return Promise.all(txHash.map(
+      oneTxHash => self.getTransactionReceiptMined(oneTxHash, interval),
+    ));
+  }
+  if (typeof txHash === 'string') {
+    return new Promise(transactionReceiptAsync);
```

Comment:
Again, why `new Promise` and not `return self.getTransactionReceiptMined(...` as in the array example?

Reply:
If `new Promise` is removed then `transactionReceiptAsync` needs to be async. 
Also`await`  is needed in  `getTransactionReceipt` statement and the awaited result can be returned. But if you see below snippet. 
```
  setTimeout(
          () => transactionReceiptAsync(resolve, reject),
          interval || 500,
        );
```
There is no sophisticated way to return result from `setTimeOut`. Adding a new promise and resolving it on receipt makes it very easy in my opinion.